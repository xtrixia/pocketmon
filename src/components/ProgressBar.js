import { css } from "@emotion/react";

import { COLORS } from "../root/colors";

function ProgressBar({ value, color }) {
  return (
    <div
      aria-valuenow={value}
      aria-valuemin={0}
      aria-valuemax={100}
      className={css`
        width: 100%;
        height: 1rem;
        position: relative;
        overflow: hidden;
        background: ${COLORS.white2};
        :before {
          display: block;
          position: absolute;
          width: 100%;
          height: 100%;
          content: "";
          background: ${color};
          left: -${100 - value}%;
        }
      `}
      role="progressbar"
    />
  );
}

export default ProgressBar;
