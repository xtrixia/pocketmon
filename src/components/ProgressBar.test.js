import React from "react";
import renderer from "react-test-renderer";

import ProgressBar from "./ProgressBar";

test("ProgressBar render correctly according to props", () => {
  const values = [{ value: 72 }, { value: 50 }];

  values.forEach((val) => {
    const component = renderer.create(
      <ProgressBar value={val.value} color={val.color}>
        Click here
      </ProgressBar>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.props.role).toStrictEqual("progressbar");
    expect(tree.props["aria-valuenow"]).toEqual(val.value);
  });
});
