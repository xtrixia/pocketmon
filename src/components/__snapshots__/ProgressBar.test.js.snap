// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar render correctly according to props 1`] = `
<div
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={72}
  className="css-1d7xzb9"
  role="progressbar"
/>
`;

exports[`ProgressBar render correctly according to props 2`] = `
<div
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={50}
  className="css-1y7tdla"
  role="progressbar"
/>
`;
